src/utils/query.utils.ts

```diff
@@ -15,47 +15,46 @@ export const IdField = 'id';
 
 /**  Converts a list of SqlWhere outlines to string of SQL clauses */
 export function compileWhere(clauses: SqlWhere[] | undefined, pagination?: SqlPaginate, order?: SqlOrder[]) {
+  if (!clauses?.length)
+      return '';
+
+  // Format clauses
+  const formattedWhere = clauses.reduce((acc, clause, index) => {
+      let value = clause.value;
+      const { isString, isNumber, isBoolean, isNull } = describeType(value);
+      const isBitwise = clause.operator === SqlWhereOperator.BitwiseAnd;
+      const isArray = [SqlWhereOperator.In, SqlWhereOperator.NotIn].includes(clause.operator);
+      // Format field
+      const field = getTargetField(FromAlias, clause.field, {
+          jsonPath: clause.jsonPath,
+          type: isNumber ? SqlType.Int : isBoolean ? SqlType.Boolean : undefined
+      });
+      // Format value
+      if (!isArray && !isNull && !isNumber && !isBoolean && !isString) {
+          value = `'${clause.value}'`;
+      }
+
+      // Add paranthesis to the last clause if paginating
+      const addParanthesis = index === clauses.length - 1 && pagination;
+
+      return `${acc}${index > 0 ? ` ${clause.andOr || AndOr.And} ` : ''}${field} ${clause.operator} ${value || ''}${isBitwise ? ' > 0' : ''}${addParanthesis ? ')' : ''}`;
+  }, `WHERE ${pagination ? '(' : ''}`);
 
   // Create a clause for the pagination cursor
   if (pagination) {
-    let operator = SqlWhereOperator.Gt;
-
-    if (order?.length) {
-      const direction = order.find((item) => item.field === pagination.field)?.direction;
-      if (direction === SqlDirection.Desc) operator = SqlWhereOperator.Lt;
-    }
-
-    clauses ||= [];
-    clauses.push({
-      field: pagination.field,
-      operator: operator,
-      value: pagination.cursor
-    });
-  }
-
-  if (!clauses?.length) return '';
+      let operator = SqlWhereOperator.Gt;
+      if (order?.length) {
+          const direction = order.find((item) => item.field === pagination.field)?.direction;
+          if (direction === SqlDirection.Desc)
+              operator = SqlWhereOperator.Lt;
+      }
 
-  // Format clauses
-  return clauses.reduce((acc, clause, index) => {
-    let value = clause.value;
-    const { isString, isNumber, isBoolean, isNull } = describeType(value);
-    const isBitwise = clause.operator === SqlWhereOperator.BitwiseAnd;
-    const isArray = [SqlWhereOperator.In, SqlWhereOperator.NotIn].includes(clause.operator);
-
-    // Format field
-    const field = getTargetField(FromAlias, clause.field, {
-      jsonPath: clause.jsonPath,
-      type: isNumber ? SqlType.Int : isBoolean ? SqlType.Boolean : undefined
-    });
+      const paginatedWhere = `${AndOr.And} ${pagination.field} ${operator} ${pagination.cursor || ''}`;
 
-    // Format value
-    if (!isArray && !isNull && !isNumber && !isBoolean && !isString) {
-      value = `'${clause.value}'`;
-    }
+      return `${formattedWhere} ${paginatedWhere}`
```

Comment:
Similarly, if there are no clauses, the result will just be `AND <pagination clause>`